### PR TITLE
NMS-12847: SslContextFactory needs to be changed to SslContextFactory.Server in jetty.xml

### PR DESCRIPTION
**Pull request by jeffgdotorg**
### All Contributors

* [x] Have you read and followed our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [x] Have you [made an issue in the OpenNMS issue tracker](https://issues.opennms.org)?<br>If so, you should:
* [x] Have you made a comment in that issue which points back to this PR?
* [x] Have you updated the JIRA link at the bottom of this comment to link to your issue?
* [ ] If this is a new or updated feature, is there documentation for the new behavior?
* [x] If this is new code, are there unit and/or integration tests?
* [x] If this PR targets a `foundation-*` branch, does it avoid changing files in `$OPENNMS_HOME/etc/`?

### Pull Request Process

One or more reviewers should be assigned to each PR.

If you know that a particular person is subject matter expert in the area your PR affects, feel free to assign one or more reviewers when you create this PR, otherwise reviewers will be assigned for you.

If you have made additions or changes to the documentation, or if you _need_ documentation for these code changes, please make sure a technical writer has looked it over.

Once the reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the Order of the Green Polo) you are welcome to merge the PR.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-12847

### CODE REVIEW

**Comment by jeffgdotorg** [comment]
Changes pass smoke.

File: core/upgrade/src/test/resources/etc/examples/jetty.xml [hunk 1]
```diff
@@ -32,7 +32,7 @@
 		<Arg>
 			<New class="org.eclipse.jetty.server.ssl.SslSelectChannelConnector">
 				<Arg>
-					<New class="org.eclipse.jetty.http.ssl.SslContextFactory">
+					<New class="org.eclipse.jetty.http.ssl.SslContextFactory$Server">
```
Comment (indigo423) [inline core/upgrade/src/test/resources/etc/examples/jetty.xml:35]: @RangerRick @jeffgdotorg Just as a heads up. It seems like this is something we cannot cleanly merge forward to 26.x. The `etc/examples/jetty.xml` has changes preventing merging forward. I've just noticed this by testing with 26.1.3 and here is what I did to get it working:

```
root@ubuntu:/usr/share/opennms/etc# diff jetty.xml examples/jetty.xml
42a43
>   <!--
56c57
<   <New id="sslContextFactory" class="org.eclipse.jetty.util.ssl.SslContextFactory$Server">
---
>   <New id="sslContextFactory" class="org.eclipse.jetty.util.ssl.SslContextFactory">
114a116
>   -->
```

I've tested with Let's Encrypt certificates in the Java Keystore file and everything worked fine.

Comment (indigo423) [reply core/upgrade/src/test/resources/etc/examples/jetty.xml:35]: Other than that 👍  from me.

**Comment by RangerRick** [comment]
Just curious, what's different in 26.x?  I thought we were the same version of Jetty from here up?

**Comment by indigo423** [comment]
> Just curious, what's different in 26.x? I thought we were the same version of Jetty from here up?
```
root@ubuntu:~# diff jetty.xml.f2017 /usr/share/opennms/etc/examples/jetty.xml > jetty-f2017-h26.1.3.patch
root@ubuntu:~# cat jetty-f2017-h26.1.3.patch
4c4
< <Configure id="FileServer" class="org.eclipse.jetty.server.Server">
---
> <Configure id="Server" class="org.eclipse.jetty.server.Server">
6,121c6,206
< 	<Call name="addConnector">
< 		<Arg>
< 			<New class="org.eclipse.jetty.server.nio.SelectChannelConnector">
< 				<Set name="statsOn">true</Set>
< 				<Set name="host"><SystemProperty name="org.opennms.netmgt.jetty.host" default="0.0.0.0" /></Set>
< 				<Set name="port"><SystemProperty name="org.opennms.netmgt.jetty.port" default="8980" /></Set>
< 				<Set name="requestHeaderSize"><SystemProperty name="org.opennms.netmgt.jetty.requestHeaderSize" default="4000" /></Set>
< 			</New>
< 		</Arg>
< 	</Call>
<
< 	<!-- Add AJP support -->
< 	<!--
< 	<Call name="addConnector">
< 		<Arg>
< 			<New class="org.eclipse.jetty.ajp.Ajp13SocketConnector">
< 				<Set name="statsOn">true</Set>
< 				<Set name="port">8981</Set>
< 			</New>
< 		</Arg>
< 	</Call>
< 	-->
<
< 	<!-- Add HTTPS support -->
< 	<!--
< 	<Call name="addConnector">
< 		<Arg>
< 			<New class="org.eclipse.jetty.server.ssl.SslSelectChannelConnector">
< 				<Arg>
< 					<New class="org.eclipse.jetty.http.ssl.SslContextFactory$Server">
< 						<Set name="keyStore"><SystemProperty name="org.opennms.netmgt.jetty.https-keystore" /></Set>
< 						<Set name="keyStorePassword"><SystemProperty name="org.opennms.netmgt.jetty.https-keystorepassword" default="changeit" /></Set>
< 						<Set name="keyManagerPassword"><SystemProperty name="org.opennms.netmgt.jetty.https-keypassword" default="changeit" /></Set>
< 						<Set name="trustStore"><SystemProperty name="org.opennms.netmgt.jetty.https-keystore" /></Set>
< 						<Set name="trustStorePassword"><SystemProperty name="org.opennms.netmgt.jetty.https-keystorepassword" default="changeit" /></Set>
< 						<Set name="ExcludeProtocols">
< 							<Array type="java.lang.String">
< 								<Item>SSLv3</Item>
< 							</Array>
< 						</Set>
< 					</New>
< 				</Arg>
< 				<Set name="host"><SystemProperty name="org.opennms.netmgt.jetty.https-host" default="0.0.0.0" /></Set>
< 				<Set name="port"><SystemProperty name="org.opennms.netmgt.jetty.https-port" default="8443" /></Set>
< 				<Set name="maxIdleTime">30000</Set>
< 				<Set name="ExcludeCipherSuites">
< 					<Array type="java.lang.String">
< 						<Item>SSL_DHE_DSS_WITH_DES_CBC_SHA</Item>
< 						<Item>SSL_DHE_RSA_EXPORT_WITH_DES40_CBC_SHA</Item>
< 						<Item>SSL_DHE_DSS_EXPORT_WITH_DES40_CBC_SHA</Item>
< 						<Item>SSL_DHE_RSA_WITH_DES_CBC_SHA</Item>
< 						<Item>SSL_RSA_EXPORT_WITH_DES40_CBC_SHA</Item>
< 						<Item>SSL_RSA_EXPORT_WITH_RC4_40_MD5</Item>
< 						<Item>SSL_RSA_WITH_3DES_EDE_CBC_SHA</Item>
< 						<Item>SSL_RSA_WITH_DES_CBC_SHA</Item>
< 						<Item>TLS_DHE_RSA_EXPORT_WITH_DES40_CBC_SHA</Item>
< 						<Item>TLS_RSA_EXPORT_WITH_DES40_CBC_SHA</Item>
< 						<Item>TLS_RSA_WITH_DES_CBC_SHA</Item>
< 						<Item>SSL_DHE_RSA_WITH_3DES_EDE_CBC_SHA</Item>
< 						<Item>TLS_DHE_RSA_WITH_AES_128_CBC_SHA</Item>
< 					</Array>
< 				</Set>
< 			</New>
< 		</Arg>
< 	</Call>
< 	-->
<
< 	<Set name="handler">
< 		<New id="Handlers" class="org.eclipse.jetty.server.handler.HandlerCollection">
< 			<Set name="handlers">
< 			 <Array type="org.eclipse.jetty.server.Handler">
< 				 <Item>
< 					 <New id="Contexts" class="org.eclipse.jetty.server.handler.ContextHandlerCollection"/>
< 				 </Item>
< 				 <Item>
< 					 <New id="DefaultHandler" class="org.eclipse.jetty.server.handler.DefaultHandler"/>
< 				 </Item>
< 			 </Array>
< 			</Set>
< 		</New>
< 	</Set>
<
< 	<Call name="addBean">
< 		<Arg>
< 			<New id="DeploymentManager" class="org.eclipse.jetty.deploy.DeploymentManager">
< 				<Set name="contexts">
< 					<Ref id="Contexts" />
< 				</Set>
< 				<Call name="setContextAttribute">
< 					<Arg>org.eclipse.jetty.server.webapp.ContainerIncludeJarPattern</Arg>
< 					<Arg>.*/[^/]*\.jar$</Arg>
< 				</Call>
< 			</New>
< 		</Arg>
< 	</Call>
<
< 	<Ref id="DeploymentManager">
< 		<Call name="addAppProvider">
< 			<Arg>
< 				<New class="org.eclipse.jetty.deploy.providers.ContextProvider">
< 					<Set name="monitoredDirName"><SystemProperty name="opennms.home" />/contexts</Set>
< 					<Set name="scanInterval">10</Set>
< 				</New>
< 			</Arg>
< 		</Call>
< 		<Call id="webappprovider" name="addAppProvider">
< 			<Arg>
< 				<New class="org.eclipse.jetty.deploy.providers.WebAppProvider">
< 					<Set name="monitoredDirName"><SystemProperty name="opennms.home" />/jetty-webapps</Set>
< 					<Set name="scanInterval">10</Set>
< 					<Set name="contextXmlDir"><SystemProperty name="opennms.home" />/contexts</Set>
< 					<Set name="extractWars">false</Set>
< 				</New>
< 			</Arg>
< 		</Call>
< 	</Ref>
---
>   <New id="httpConfig" class="org.eclipse.jetty.server.HttpConfiguration">
>      <Set name="requestHeaderSize"><SystemProperty name="org.opennms.netmgt.jetty.requestHeaderSize" default="4000" /></Set>
>   </New>
>
>   <Call name="addConnector">
>     <Arg>
>       <New id="httpConnector" class="org.eclipse.jetty.server.ServerConnector">
>         <Arg name="server"><Ref refid="Server" /></Arg>
>         <Arg name="factories">
>           <Array type="org.eclipse.jetty.server.ConnectionFactory">
>             <Item>
>               <New class="org.eclipse.jetty.server.HttpConnectionFactory">
>                 <Arg name="config"><Ref refid="httpConfig" /></Arg>
>                 <Arg name="compliance"><Call class="org.eclipse.jetty.http.HttpCompliance" name="valueOf"><Arg><Property name="jetty.http.compliance" default="RFC7230"/></Arg></Call></Arg>
>               </New>
>             </Item>
>           </Array>
>         </Arg>
>         <Set name="host"><SystemProperty name="org.opennms.netmgt.jetty.host" default="0.0.0.0" /></Set>
>         <Set name="port"><SystemProperty name="org.opennms.netmgt.jetty.port" default="8980" /></Set>
>       </New>
>     </Arg>
>   </Call>
>
>   <!-- Add AJP support -->
>   <!--
>   <Call name="addConnector">
>     <Arg>
>       <New class="org.eclipse.jetty.ajp.Ajp13SocketConnector">
>         <Set name="statsOn">true</Set>
>         <Set name="port">8981</Set>
>       </New>
>     </Arg>
>   </Call>
>   -->
>
>   <!-- Add HTTPS support -->
>   <!--
>   <New id="sslHttpConfig" class="org.eclipse.jetty.server.HttpConfiguration">
>     <Arg><Ref refid="httpConfig"/></Arg>
>     <Call name="addCustomizer">
>       <Arg>
>         <New class="org.eclipse.jetty.server.SecureRequestCustomizer">
>           <Arg name="sniHostCheck" type="boolean"><Property name="jetty.ssl.sniHostCheck" default="true"/></Arg>
>           <Arg name="stsMaxAgeSeconds" type="int"><Property name="jetty.ssl.stsMaxAgeSeconds" default="-1"/></Arg>
>           <Arg name="stsIncludeSubdomains" type="boolean"><Property name="jetty.ssl.stsIncludeSubdomains" default="false"/></Arg>
>         </New>
>       </Arg>
>     </Call>
>   </New>
>
>   <New id="sslContextFactory" class="org.eclipse.jetty.util.ssl.SslContextFactory">
>     <Set name="KeyStorePath"><SystemProperty name="org.opennms.netmgt.jetty.https-keystore" /></Set>
>     <Set name="KeyStorePassword"><SystemProperty name="org.opennms.netmgt.jetty.https-keystorepassword" default="changeit" /></Set>
>     <Set name="KeyStoreType"><SystemProperty name="org.opennms.netmgt.jetty.https-keystoretype" default="JKS" /></Set>
>     <Set name="KeyStoreProvider"><SystemProperty name="org.opennms.netmgt.jetty.https-keystoreprovider" /></Set>
>     <Set name="KeyManagerPassword"><SystemProperty name="org.opennms.netmgt.jetty.https-keypassword" default="changeit" /></Set>
>     <Set name="TrustStorePath"><SystemProperty name="org.opennms.netmgt.jetty.https-keystore" /></Set>
>     <Set name="TrustStorePassword"><SystemProperty name="org.opennms.netmgt.jetty.https-keystorepassword" default="changeit" /></Set>
>     <Set name="TrustStoreType"><SystemProperty name="org.opennms.netmgt.jetty.https-keystoretype" default="JKS" /></Set>
>     <Set name="TrustStoreProvider"><SystemProperty name="org.opennms.netmgt.jetty.https-keystoreprovider" /></Set>
>     <Set name="certAlias"><SystemProperty name="org.opennms.netmgt.jetty.https-cert-alias" /></Set>
>     <Set name="useCipherSuitesOrder"><Property name="jetty.sslContext.useCipherSuitesOrder" default="true"/></Set>
>     <Set name="sslSessionCacheSize"><Property name="jetty.sslContext.sslSessionCacheSize" default="-1"/></Set>
>     <Set name="sslSessionTimeout"><Property name="jetty.sslContext.sslSessionTimeout" default="-1"/></Set>
>     <Set name="ExcludeCipherSuites">
>       <Array type="java.lang.String">
>         <Item>SSL_DHE_DSS_EXPORT_WITH_DES40_CBC_SHA</Item>
>         <Item>SSL_DHE_DSS_WITH_DES_CBC_SHA</Item>
>         <Item>SSL_DHE_RSA_EXPORT_WITH_DES40_CBC_SHA</Item>
>         <Item>SSL_DHE_RSA_WITH_3DES_EDE_CBC_SHA</Item>
>         <Item>SSL_DHE_RSA_WITH_DES_CBC_SHA</Item>
>         <Item>SSL_RSA_EXPORT_WITH_DES40_CBC_SHA</Item>
>         <Item>SSL_RSA_EXPORT_WITH_RC4_40_MD5</Item>
>         <Item>SSL_RSA_WITH_3DES_EDE_CBC_SHA</Item>
>         <Item>SSL_RSA_WITH_DES_CBC_SHA</Item>
>         <Item>TLS_DHE_RSA_EXPORT_WITH_DES40_CBC_SHA</Item>
>         <Item>TLS_DHE_RSA_WITH_3DES_EDE_CBC_SHA</Item>
>         <Item>TLS_DHE_RSA_WITH_AES_128_CBC_SHA</Item>
>         <Item>TLS_RSA_EXPORT_WITH_DES40_CBC_SHA</Item>
>         <Item>TLS_RSA_WITH_DES_CBC_SHA</Item>
>      </Array>
>     </Set>
>   </New>
>
>   <Call name="addConnector">
>     <Arg>
>       <New id="sslConnector" class="org.eclipse.jetty.server.ServerConnector">
>         <Arg name="server"><Ref refid="Server" /></Arg>
>         <Arg name="factories">
>           <Array type="org.eclipse.jetty.server.ConnectionFactory">
>             <Item>
>               <New class="org.eclipse.jetty.server.SslConnectionFactory">
>                 <Arg name="next">http/1.1</Arg>
>                 <Arg name="sslContextFactory"><Ref refid="sslContextFactory"/></Arg>
>               </New>
>             </Item>
>             <Item>
>               <New class="org.eclipse.jetty.server.HttpConnectionFactory">
>                 <Arg name="config"><Ref refid="sslHttpConfig" /></Arg>
>                 <Arg name="compliance"><Call class="org.eclipse.jetty.http.HttpCompliance" name="valueOf"><Arg><Property name="jetty.http.compliance" default="RFC7230"/></Arg></Call></Arg>
>               </New>
>             </Item>
>           </Array>
>         </Arg>
>         <Set name="host"><SystemProperty name="org.opennms.netmgt.jetty.https-host" default="0.0.0.0" /></Set>
>         <Set name="port"><SystemProperty name="org.opennms.netmgt.jetty.https-port" default="8443" /></Set>
>       </New>
>     </Arg>
>   </Call>
>   -->
>
>   <!-- Settings for X-Frame-Options to avoid clickjacking -->
>   <!-- Comment out this block and the below reference to RewriteHandler to allow framing -->
>   <New id="RewriteHandler" class="org.eclipse.jetty.rewrite.handler.RewriteHandler">
>     <Set name="rules">
>       <Array type="org.eclipse.jetty.rewrite.handler.Rule">
>         <Item>
>           <New id="header" class="org.eclipse.jetty.rewrite.handler.HeaderPatternRule">
>             <Set name="pattern">*</Set>
>             <Set name="name">X-Frame-Options</Set>
>             <Set name="value">SAMEORIGIN</Set>
>           </New>
>         </Item>
>       </Array>
>     </Set>
>   </New>
>
>   <!-- Enable statistics -->
>   <Call name="insertHandler">
>     <Arg>
>       <New id="StatsHandler" class="org.eclipse.jetty.server.handler.StatisticsHandler">
>       </New>
>     </Arg>
>   </Call>
>   <Call class="org.eclipse.jetty.server.ConnectorStatistics" name="addToAllConnectors">
>     <Arg><Ref refid="Server"/></Arg>
>   </Call>
>
>   <New id="Handlers" class="org.eclipse.jetty.server.handler.HandlerCollection">
>       <Set name="handlers">
>        <Array type="org.eclipse.jetty.server.Handler">
>            <!-- Comment out this item to allow framing -->
>            <Item>
>              <Ref id="RewriteHandler"/>
>            </Item>
>            <Item>
>                <New id="Contexts" class="org.eclipse.jetty.server.handler.ContextHandlerCollection"/>
>            </Item>
>            <Item>
>                <New id="DefaultHandler" class="org.eclipse.jetty.server.handler.DefaultHandler"/>
>            </Item>
>          <!-- NCSA Request Logging
>            <Item>
>                <New id="RequestLog" class="org.eclipse.jetty.server.handler.RequestLogHandler">
>                  <Set name="requestLog">
>                    <New id="RequestLogImpl" class="org.eclipse.jetty.server.NCSARequestLog">
>                      <Arg>logs/jetty-requests-yyyy_mm_dd.log</Arg>
>                      <Set name="retainDays">90</Set>
>                      <Set name="append">true</Set>
>                      <Set name="extended">true</Set>
>                      <Set name="logTimeZone">US/Central</Set>
>                    </New>
>                  </Set>
>                </New>
>            </Item>
>          -->
>        </Array>
>       </Set>
>   </New>
>
>   <Set name="handler">
>     <New id="mdcHandler" class="org.opennms.netmgt.jetty.MDCHandler">
>       <Set name="handler"><Ref id="Handlers" /></Set>
>     </New>
>   </Set>
>
>   <New id="DeploymentManager" class="org.eclipse.jetty.deploy.DeploymentManager">
>     <Set name="contexts">
>       <Ref id="Contexts" />
>     </Set>
>     <Call name="setContextAttribute">
>       <Arg>org.eclipse.jetty.server.webapp.ContainerIncludeJarPattern</Arg>
>       <Arg>.*/[^/]*\.jar$</Arg>
>     </Call>
>     <Call name="addAppProvider">
>       <Arg>
>         <New class="org.opennms.netmgt.jetty.OpenNMSWebAppProvider">
>           <Set name="monitoredDirName"><SystemProperty name="opennms.home" />/jetty-webapps</Set>
>           <Set name="scanInterval">10</Set>
>           <Set name="extractWars">false</Set>
>         </New>
>       </Arg>
>     </Call>
>   </New>
>
>   <Call name="addBean">
>     <Arg>
>       <Ref refid="DeploymentManager"/>
>     </Arg>
>   </Call>
```

**Comment by RangerRick** [comment]
Do it!

**Comment by RangerRick** [comment]
I went ahead and made [a branch](https://app.circleci.com/pipelines/github/OpenNMS/opennms/7442/workflows/d5b11f3c-38ea-43ba-b95b-9ad888b88684) with @indigo423's changes against `foundation-2020` just to make sure it goes green.  Ronny, is it correct that _both_ files need changing?

**Comment by indigo423** [comment]
> I went ahead and made [a branch](https://app.circleci.com/pipelines/github/OpenNMS/opennms/7442/workflows/d5b11f3c-38ea-43ba-b95b-9ad888b88684) with @indigo423's changes against `foundation-2020` just to make sure it goes green. Ronny, is it correct that _both_ files need changing?

I've used the example file. I've no idea where the other one comes from to be honest. The `jetty.xml` doesn't exist by default. Pretty good question :)

**Comment by RangerRick** [comment]
Yeah I really don't know; thought maybe it was the default one inside the jar or something.  Wonder if we could just... delete it?  A question for another issue.  :D

**Pull request merged**